src/main/scala/org/neo4j/spark/util/Validations.scala

```diff
@@ -94,6 +95,7 @@ object Validations {
     val cache = new DriverCache(neo4jOptions.connection, jobId)
     val schemaService = new SchemaService(neo4jOptions, cache)
```

Comment:
A read is always tied to a set of options.

You have validateConnection called twice, in reader and in writer.   Rather than remembering to call it each time, if a SchemaService is tied to a set of options, you could put this into the initialization of the schemaservice.  Before you even get to the query, the schemaservice could validate connection for you, and the code would maybe be more dry, you don't have to remember to validate each time.

Reply:
True what you said, also @conker84 made the same observation. But I would rather validate the connection in the Validator than validate in the SchemaService. Yep the code will be more dry, but if I need to find the connection validation, the validator would be the first file I would look into.

I don't really have a strong opinion on this. I could change it if the majority agrees it's better the other way, I was just explaining why it is how it is